feat: add inheritance view to ssp-generate and ssp-assemble

jpower432 opened the pull request:
## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [X] New feature (non-breaking change which adds functionality)
- [X] Documentation (change which affects the documentation site)

## Quality assurance (all should be covered).

- [X] My code follows the code style of this project.
- [X] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [X] All new and existing tests passed.
- [X] All commits are signed-off.

## Summary

Adds an inheritance view to the generated trestle markdown that allows exported provided and responsibility statements to from a leveraged SSP to be used.

Closes #1398 
Closes #1393


- [X] Filter written controls in inheritance view by profile passed to `ssp-generate`
- [X] Ensure unimplemented statements are present after `ssp-assemble`

### Assumptions/Constraints 

- This assumes that the control exists within the `control implementation` of the SSP and edits the existing implemented requirements. It will edit or add by-component assemblies with the information.
- The `responsibility` uuid can only occur once in the `satisfied` array and the `provided` uuid can only occur once in the `inherited` array.
- This assumes that if no leveraging components are mapped to a leveraged component that the authorization is not leveraged (i.e. the `leveraged authorization` section will not be present)

### What this does not do

> Requesting to iterate in follow on PRs

- This does not does remove files in the markdown directory for provided or responsibility exported statements with changed uuids (i.e. does not delete exports no longer referenced)
- This does not accept more than one leveraged ssp

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

jpower432 commented:
@AleJo2995 @vikas-agarwal76 Added you both as reviewers to get your thoughts on this approach. We added enough here to address the two linked issues and get to the point where we can complete additional testing, so the scope of this feature should not increase. Thanks in advance!

degenaro commented:
Perhaps these pre-existed, but osnar says there are 3 code smells https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&sinceLeakPeriod=true&pullRequest=1441&id=compliance-trestle

Not essential that these get fixed, but if would be nice...

jpower432 commented on trestle/core/commands/author/ssp.py:
```diff
@@ -519,10 +563,18 @@ def _run(self, args: argparse.Namespace) -> int:
                     raise TrestleError('Original ssp has no system component.')
                 comp_dict[const.SSP_MAIN_COMP_NAME] = sys_comp
 
+                ssp_sys_imp_comps = ssp.system_implementation.components
```
@vikas-agarwal76 Just wanted to call out this logic specifically.  This is really the only change to `ssp-assemble` that was made though it would only apply to a leveraging SSP. All other changes are processed conditionally when the `leveraged-ssp` flag is used (`generate`) or the inheritance markdown is present (`assemble`).

vikas-agarwal76 reviewed (changes requested):
Small changes.

jpower432 commented:
> Perhaps these pre-existed, but sonar says there are 3 code smells https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&sinceLeakPeriod=true&pullRequest=1441&id=compliance-trestle
> 
> Not essential that these get fixed, but it would be nice...

@degenaro Working through these. Only one was pre-existing.

jpower432 commented:
@vikas-agarwal76 Changes addressing your feedback added on this commit - https://github.com/IBM/compliance-trestle/pull/1441/commits/a10ba54267798f17d201876bc6c407183f2ff48e. Please let me know if there are any additional changes or feedback.

vikas-agarwal76 reviewed (approved):
LGTM

jpower432 commented:
Thanks @vikas-agarwal76. Just wanted to confirm with you, @AleJo2995, and @degenaro, is there any other actions that need to be taken before merge? Thanks in advance!
